jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandlerCollection.java

```diff
@@ -228,22 +209,26 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         }
         else
         {
-            // This may not work in all circumstances... but then I think it should never be called
-            for (int i=0;i<handlers.length;i++)
+            if (mapping._handlers==null)
+                return;
+            for (int i=0;i<mapping._handlers.length;i++)
             {
-                handlers[i].handle(target,baseRequest, request, response);
+                mapping._handlers[i].handle(target,baseRequest, request, response);
                 if ( baseRequest.isHandled())
                     return;
             }
         }
     }
 
     /* ------------------------------------------------------------ */
-    /** Add a context handler.
+    /**
+     * Adds a context handler.
+     *
      * @param contextPath  The context path to add
      * @param resourceBase the base (root) Resource
      * @return the ContextHandler just added
      */
+    @Deprecated
     public ContextHandler addContext(String contextPath,String resourceBase)
```

Comment:
Why has this method been deprecated?
Can you please add javadoc `@deprecated` telling what to use instead?

Reply:
lots of deprecations here, gotta remember to remove them from the codebase when merging up to jetty-10.0.x